### PR TITLE
Gcc linux warnings removal

### PR DESCRIPTION
**Pull request by adrien-lessard**
Removes warnings when compiling on Linux with gcc. Some fixes are debatable and could be done in other ways. See comments in code for details.

System used:

```
$gcc --version
gcc (Ubuntu 11.3.0-1ubuntu1~22.04) 11.3.0
Copyright (C) 2021 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

$uname -a
Linux 5.10.16.3-microsoft-standard-WSL2 #1 SMP Fri Apr 2 22:23:49 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux
```

Warnings cleared:

[warnings.txt](https://github.com/bbaggerman/irig106lib/files/10375709/warnings.txt)

### CODE REVIEW

File: gcc/Makefile [hunk 1]
```diff
@@ -6,7 +6,7 @@ ifeq ($(GCC),)
 GCC := gcc
 endif
 
-CFLAGS=-D_FILE_OFFSET_BITS=64 -D_LARGEFILE64_SOURCE -ggdb -fPIC -DIRIG_NETWORKING -DSHA256ENABLE -Wall
+CFLAGS=-D_FILE_OFFSET_BITS=64 -D_LARGEFILE64_SOURCE -ggdb -fPIC -DIRIG_NETWORKING -DSHA256ENABLE -Wall -Wno-address-of-packed-member
```
Comment (adrien-lessard) [inline gcc/Makefile:9]: We accept that we will systematically be adressing structures without proper alignment. Most compilers will handle this behavior correctly.

File: src/i106_data_stream.c [hunk 2]
```diff
@@ -670,6 +670,9 @@ static EnI106Status
 #endif
             break;
 
+        default :
+            break;
+
```
Comment (adrien-lessard) [inline src/i106_data_stream.c:675]: Handle default cases for each switch

File: src/i106_data_stream.c [hunk 3]
```diff
@@ -1274,6 +1280,7 @@ EnI106Status I106_CALL_DECL
         enReturnStatus = I106_WRITE_ERROR;
 #else
 // TODO - LINUX CODE
+    (void)suUdpHeaderNonF1Seg;
```
Comment (adrien-lessard) [inline src/i106_data_stream.c:1283]: Remove unused variable warning for variables that are used in windows but nor linux or vice versa

File: src/i106_decode_16pp194.c [hunk 4]
```diff
@@ -74,7 +74,6 @@ namespace Irig106 {
  * --------------------
  */
 
-static void vFillInMsgPtrs(Su16PP194_CurrMsg * psuCurrMsg);
```
Comment (adrien-lessard) [inline src/i106_decode_16pp194.c:77]: Unused function

File: src/i106_decode_tmats.c [hunk 5]
```diff
@@ -185,7 +185,6 @@ EnI106Status I106_CALL_DECL
     int                 iCodeNameLength;
     char              * szCodeName;
     char              * szDataItem;
-    int                 bParseError;
```
Comment (adrien-lessard) [inline src/i106_decode_tmats.c:188]: Unused error result

File: src/i106_index.c [hunk 6]
```diff
@@ -465,7 +465,6 @@ EnI106Status ProcessNodeIndexPacket(int iHandle, int64_t lNodeIndexOffset)
 void AddIndexNodeToIndex(int iHandle, SuIndex_CurrMsg * psuNodeIndexMsg, uint16_t uChID, uint8_t ubyDataType)
     {
     SuPacketIndexInfo   suIndexInfo;
-    EnI106Status        enStatus;
```
Comment (adrien-lessard) [inline src/i106_index.c:468]: unused status

File: src/irig106ch10.c [hunk 7]
```diff
@@ -36,9 +36,13 @@
  ****************************************************************************/
 
 #if defined(__GNUC__)
+#ifndef _FILE_OFFSET_BITS
 #define _FILE_OFFSET_BITS 64
```
Comment (adrien-lessard) [inline src/irig106ch10.c:40]: redefinition of _FILE_OFFSET_BITS

File: src/irig106ch10.c [hunk 8]
```diff
@@ -597,7 +598,7 @@ EnI106Status I106_CALL_DECL
 #if defined(IRIG_NETWORKING)
             else
                 enI106_MoveReadPointer(iHandle, (long)llSkipSize);
-                break;
+            break;
```
Comment (adrien-lessard) [inline src/irig106ch10.c:601]: Misleading indentation

File: src/irig106ch10.c [hunk 9]
```diff
@@ -696,7 +700,7 @@ EnI106Status I106_CALL_DECL
                 switch (g_suI106Handle[iHandle].enFileMode)
                     {
                     case I106_READ :
-                        iReadCnt = read(g_suI106Handle[iHandle].iFile, &psuHeader->abyTime[0], SEC_HEADER_SIZE);
+                        iReadCnt = read(g_suI106Handle[iHandle].iFile, psuHeader->abySecHdr, SEC_HEADER_SIZE);
```
Comment (adrien-lessard) [inline src/irig106ch10.c:703]: Out of bound access

File: src/irig106ch10.c [hunk 10]
```diff
@@ -1686,7 +1700,7 @@ this routine will work fine. But a filler value of 0xFF legal (but not common).
 case this routine will give incorrect results.
 */
 uint32_t I106_CALL_DECL
-    uCalcDataChecksum(int iChecksumType, void * pvBuff, uint32_t uBuffLen, uint32_t uChecksum)
+    uCalcDataChecksum(int iChecksumType, const void * pvBuff, uint32_t uBuffLen, uint32_t uChecksum)
```
Comment (adrien-lessard) [inline src/irig106ch10.c:1703]: Compile error, mismatch with function prototype, see https://github.com/bbaggerman/irig106lib/pull/27

File: src/irig106ch10.h [hunk 11]
```diff
@@ -228,9 +228,16 @@ typedef PUBLIC struct SuI106Ch10Header_S
...
+    union
+        {
+        struct
+            {
+            uint8_t       abyTime[8];   ///< Time (start secondary header)
+            uint16_t      uReserved;    //
+            uint16_t      uSecChecksum; ///< Secondary Header Checksum
+            };
+        uint8_t       abySecHdr[12];    ///< Secondary Header
+        };
```
Comment (adrien-lessard) [inline src/irig106ch10.h:240]: Union type to prevent out of bound access warning

**Review by bbaggerman — APPROVED** [review]

**Review by bbaggerman — APPROVED** [review]

**Pull request merged**